base/src/main/java/org/aya/tyck/pat/PatTycker.java

```diff
@@ -245,40 +244,41 @@ private Pat.Preclause<Term> checkRhs(LhsResult lhsResult) {
       ? new ErrorTerm(e, false)
       : exprTycker.inherit(new BodySubstitutor(e.view(), lhsResult.bodySubst).commit(), type).wellTyped());
     exprTycker.localCtx = parent;
-    return new Pat.Preclause<>(lhsResult.preclause.sourcePos(), patterns, term);
+    return new Pat.Preclause<>(lhsResult.preclause.sourcePos(), lhsResult.preclause.patterns(), term);
   }
 
   private LhsResult checkLhs(Pattern.Clause match, Def.Signature signature) {
     var parent = exprTycker.localCtx;
     exprTycker.localCtx = parent.deriveMap();
     currentClause = match;
     var step0 = visitPatterns(signature, match.patterns.view());
+    var patterns = step0._1.map(p -> p.inline(exprTycker.localCtx)).toImmutableSeq();
     var step1 = new LhsResult(exprTycker.localCtx, step0._2, bodySubst.toImmutableMap(), match.hasError,
```

Comment:
Just random thoughts: can we `localCtx.put(fakeBind, term)` in `visitPatterns` ([here](https://github.com/aya-prover/aya-dev/blob/main/base/src/main/java/org/aya/tyck/pat/PatTycker.java#L197))? I see the new `inline` just did the same thing.

Reply:
There are more than one place where you mutate Pat.Meta::solution.

Reply:
I see!! Thank you~